geomesa-spark/geomesa-spark-jts/src/main/scala/org/apache/spark/sql/jts/AbstractGeometryUDT.scala

```diff
@@ -23,21 +20,11 @@ import scala.reflect._
  */
 abstract class AbstractGeometryUDT[T >: Null <: Geometry: ClassTag](override val simpleString: String)
   extends UserDefinedType[T] {
+  protected def gf: GeometryFactory = AbstractGeometryUDT.gf
+  override lazy val pyUDT: String = s"geomesa_pyspark.types.${getClass.getSimpleName}"
+  override lazy val userClass: Class[T] = classTag[T].runtimeClass.asInstanceOf[Class[T]]
+}
 
-  override def pyUDT: String = s"geomesa_pyspark.types.${getClass.getSimpleName}"
-
-  override def serialize(obj: T): InternalRow = {
-    new GenericInternalRow(Array[Any](WKBUtils.write(obj)))
-  }
-
-  override def sqlType: DataType = StructType(Seq(
```

Comment:
I think this becomes problematic later, because the way the UDT is declared on the python side does not match. See [here](https://github.com/elahrvivaz/geomesa/blob/fcr_spark_sql_geom_format/geomesa-spark/geomesa_pyspark/src/main/python/geomesa_pyspark/types.py#L10-L12), no changes.

Reply:
Thanks for linking to that.  Emilio and I are a little rough on how the Python side of this integration works...

Reply:
yeah, thanks for looking at this! I will try to figure out and resolve the python discrepancies.